broker/fragment/store_azure.go

```diff
@@ -327,7 +327,11 @@ func (a *azureBackend) buildBlobURL(cfg AzureStoreConfig, client pipeline.Pipeli
 func (a *azureBackend) getUserDelegationCredential() (*service.UserDelegationCredential, error) {
 	// We want to make sure we create a new credential well before the existing one expires
 	// So this gives us a 60 second buffer before the credential expires to make a new one.
-	if a.udc == nil || (a.udcExp != nil && a.udcExp.After(time.Now().Add(time.Minute))) {
+
```

Comment:
nit: extra newline.

nit: I think at least an hour, perhaps even more would be appropriate. Otherwise you can get a signed fragment with a larger requested TTL that's invalidated far before that actual TTL due to the signing credential itself expiring.

If we wanted to get _really_ fancy then we ought to be checking the expiry in comparison to a stamped TTL, but :shrug:

Reply:
Ok, I bumped it all the way up to 5 days minimum UDC window. Microsoft says it maxes out at 7, do what about generating UDCs that last for 6 days, and refreshing once the cached credential has less than 5d left on its timer, so we're always signing URLs with a credential that has at least 5 days left on the clock?